pkg/worker/events.go

```diff
@@ -0,0 +1,97 @@
+package worker
+
+import (
+	"log"
+	"time"
+
+	"github.com/beam-cloud/beta9/pkg/repository"
+	types "github.com/beam-cloud/beta9/pkg/types"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func collectAndSendContainerMetrics(eventRepo repository.EventRepository, workerID string, request *types.ContainerRequest, spec *specs.Spec, pidChan <-chan int, done chan bool) {
+	gpuEnabled := request.Gpu != ""
+
+	ticker := time.NewTicker(types.ContainerResourceUsageEmissionInterval)
+	defer ticker.Stop()
+
+	containerPid := <-pidChan
+	if containerPid == 0 {
+		return
+	}
+
+	containerProc, err := NewProcUtil(containerPid)
+	if err != nil {
+		log.Printf("Unable to get process for container %s: %v\n", request.ContainerId, err)
+		return
+	}
+
+	var prevProcCpuTime float64 = 0
+	var prevSystemCpuTime float64 = 0
+	for {
+		select {
+		case <-done:
+			return
+
+		case <-ticker.C:
+			// Calculate GPU memory
+			var gpuMemUsed int64
+			var gpuMemTotal int64
+			if gpuEnabled {
+				for _, device := range spec.Linux.Resources.Devices {
```

Comment:
I know it's difficult to test locally but it seems like in the old repo we had a check for `nvmlActive`. What happened to that?

Reply:
We got rid of the nvml dependency and opted for just using nvidia-smi instead. In v1, we had nvml and we had to have different go files per operating system e.g. darwin and linux. Because we develop on macos, we didn't get intellisense/code completion or static typing since we're developing for the linux side.

Reply:
makes sense